hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/BulkInsertPartitioner.java

```diff
@@ -72,4 +78,31 @@ default Option<WriteHandleFactory> getWriteHandleFactory(int partitionId) {
     return Option.empty();
   }
 
+  /*
+   * If possible, we want to sort the data by partition path. Doing so will reduce the number of files written.
+   **/
+  static String[] prependPartitionPathColumn(String[] columnNames, HoodieWriteConfig config) {
+    String partitionPath = config.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key());
+    if (partitionPath.isEmpty() || config.getMetadataConfig().populateMetaFields()) {
```

Comment:
When meta-fields are enabled, we should use `PARTITION_PATH_METADATA_FIELD` instead of `KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME`